addons/hr_material/models/hr_material.py

```diff
@@ -0,0 +1,342 @@
+# -*- coding: utf-8 -*-
+from openerp import models, fields, api, _, tools
+from openerp.exceptions import Warning
+
+
+class HrMaterialStage(models.Model):
+
+    """Model for case stages. This models the main stages of a Maintenance Request management flow.
+       Stages are for example used to display the kanban view of records.
+    """
+
+    _name = 'hr.material.stage'
+    _description = 'Material Stage'
+    _order = 'sequence asc'
+
+    name = fields.Char('Name', required=True, translate=True)
+    sequence = fields.Integer('Sequence')
+    fold = fields.Boolean('Folded in Kanban View')
+
+
+class HrMaterialCategory(models.Model):
+    _name = 'hr.material.category'
+    _inherits = {"mail.alias": "alias_id"}
+    _inherit = ['mail.thread']
+    _description = 'Material Category'
+
+    @api.one
+    def _compute_material_count(self):
+        self.material_count = len(self.material_ids)
+        self.fold = False if self.material_count else True
+
+    @api.one
+    def _compute_maintenance_count(self):
+        self.maintenance_count = len(self.maintenance_ids)
+
+    name = fields.Char('Category Name', required=True, translate=True)
+    user_id = fields.Many2one('res.users', 'Responsible', track_visibility='onchange', default=lambda self: self.env.uid)
+    color = fields.Integer('Color Index')
+    note = fields.Text('Comments', translate=True)
+    material_ids = fields.One2many('hr.material', 'category_id', string='Materials', copy=False)
+    material_count = fields.Integer(compute='_compute_material_count', string="Material")
+    maintenance_ids = fields.One2many('hr.material.request', 'category_id', copy=False, domain=[('stage_id.fold', '=', False)])
+    maintenance_count = fields.Integer(compute='_compute_maintenance_count', string="Maintenance")
+    alias_id = fields.Many2one(
+        'mail.alias', 'Alias', ondelete="restrict", required=True,
+        help="Email alias for this material category. New emails will automatically "
+        "create new maintenance request for this material category.")
+    fold = fields.Boolean(compute='_compute_material_count', string='Folded in Kanban View')
+
+    @api.model
+    def create(self, vals):
+        self = self.with_context(alias_model_name='hr.material.request', alias_parent_model_name=self._name)
+        category_id = super(HrMaterialCategory, self).create(vals)
+        category_id.alias_id.write({'alias_parent_thread_id': category_id.id, 'alias_defaults': {'category_id': category_id.id}})
+        return category_id
+
+    @api.multi
+    def unlink(self):
+        alias_ids = []
+        for category in self:
+            if category.material_ids or category.maintenance_ids:
+                raise Warning(_('You cannot delete a Material Category containing Material/Maintenance Request. You can either delete all the material/maintenance which belongs to in this category and then delete the material category.'))
+            elif category.alias_id:
```

Comment:
What's the purpose of this condition ? aliases should be deleted with an ondelete=cascade, nothing to check.